src/machine/machined-dbus.c

```diff
@@ -288,23 +288,15 @@ static int method_create_or_register_machine(
                 return sd_bus_error_set(error, SD_BUS_ERROR_INVALID_ARGS, "Root directory must be empty or an absolute path");
 
         if (leader == 0) {
-                _cleanup_(sd_bus_creds_unrefp) sd_bus_creds *creds = NULL;
-
-                r = sd_bus_query_sender_creds(message, SD_BUS_CREDS_PID, &creds);
+                r = bus_query_sender_pidref(message, &pidref);
```

Comment:
Doesn't this require a fallback in case the option is not available?

Reply:
Hmm? What option specifically?

Reply:
for the pidfd handling

Reply:
there's already code in place to handle pidfd being unavailable. see bus_creds_get_pidref()